policyengine_us/variables/gov/states/mo/tax/income/credits/property_tax/mo_ptc_gross_income.py

```diff
@@ -0,0 +1,73 @@
+from policyengine_us.model_api import *
+
+
+class mo_ptc_gross_income(Variable):
+    value_type = float
+    entity = TaxUnit
+    label = "MO property tax credit gross income"
+    unit = USD
+    definition_period = YEAR
+    reference = (
+        "https://dor.mo.gov/forms/MO-PTS_2021.pdf",
+        "https://dor.mo.gov/forms/4711_2021.pdf",
+        "https://revisor.mo.gov/main/OneSection.aspx?section=135.010&bid=6435&hl=property+tax+credit%u2044",
+    )
+    defined_for = StateCode.MO
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        # compute core person-level income included in PTC gross income
+        core_income = tax_unit.sum(
+            add(
+                person,
+                period,
+                [
+                    "mo_adjusted_gross_income",  # Form MO-PTS, line 1
+                    "tax_exempt_pension_income",  # Form MO-PTS, line 3
+                    "tax_exempt_interest_income",  # Form MO-PTS, line 3
+                ],
+            )
+        )
+        # compute core taxunit-level income included in PTC gross income
+        tax_exempt_socsec_benefits = tax_unit(
+            "tax_exempt_social_security",  # Form MO-PTS, line 2
+            period,
+        )
+        # compute veterans benefits included in PTC gross income
+        veterans_benefits = person("veterans_benefits", period)
+        exclude = person("is_fully_disabled_service_connected_veteran", period)
+        included_veterans_benefits = ~exclude * veterans_benefits
+        veterans_benefits_income = tax_unit.sum(
+            # Form MO-PTS, line 5
+            included_veterans_benefits
+        )
+        # compute person-level public assistance included in PTC gross income
+        public_assistance_income = tax_unit.sum(
```

Comment:
Could you make this a separate variable based on a list parameter? Optionally break up some of the other pieces, as this formula is getting a bit long.

Reply:
Done